pkg/asset/manifests/gcp/cluster.go

```diff
@@ -1 +1,150 @@
 package gcp
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	capg "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+)
+
+const (
+	// autoCreateSubnets is not used as the subnet information is provided through the install config.
+	autoCreateSubnets = false
+
+	defaultSubnetCreationPurpose = "PRIVATE_RFC_1918"
+)
+
+var (
+	subnetName = func(infraID, machineType string) string { return fmt.Sprintf("%s-%s-subnet", infraID, machineType) }
+)
+
+// GenerateClusterAssets generates the manifests for the cluster-api.
+func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID *installconfig.ClusterID) (*capiutils.GenerateClusterAssetsOutput, error) {
+	manifests := []*asset.RuntimeFile{}
+
+	networkName := installConfig.Config.GCP.Network
+	if installConfig.Config.GCP.NetworkProjectID != "" {
+		// Subnets should exist in the network with name NetworkProjectID when provided.
+		networkName = installConfig.Config.GCP.NetworkProjectID
```

Comment:
This is assigning the Host Network project ID to the name of the network. These should be different. For example 
```yaml
    network: installer-shared-vpc
    networkProjectID: openshift-dev-installer
```